storage/redis_controller.go

```diff
@@ -136,26 +137,28 @@ func (rc *RedisController) ConnectToRedis(ctx context.Context, onReconnect func(
 			RedisController: rc,
 		},
 		{
-			IsCache:         true,
 			RedisController: rc,
+			IsCache:         true,
 		},
 		{
-			IsAnalytics:     true,
 			RedisController: rc,
+			IsAnalytics:     true,
 		},
 	}
 
 	// First time connecting to the clusters. We need this for the first connection (and avoid waiting 1second for the rc.statusCheck loop).
 	up := true
 	for _, v := range c {
-		if !rc.connectSingleton(v.IsCache, v.IsAnalytics, *conf) {
-			up = false
-			break
-		}
-
-		if !clusterConnectionIsOpen(&v) {
-			up = false
-			break
+		rc.connectSingleton(v.IsCache, v.IsAnalytics, *conf)
+		err := backoff.Retry(func() error {
+			isOpen := clusterConnectionIsOpen(&v)
```

Comment:
@mativm02 just to validate - will this retry to connect or retry to check if it's connected?

Reply:
It will retry to check if it's connected. It's not needed to retry the connection, since the instance is already created (even if the Redis instance is down when starting the gateway, the Redis client will be initialized). We just need to make sure it's working, and if it's not, we'll log an error message.
AFAIK, It's not like Mongo for example, which tries to connect to the Mongo instance and if it wasn't successful, it returns an error.